<?xml version="1.0" ?>
<sdf version="1.4">
  <world name="default">
    <plugin name="ros_link_attacher_plugin" filename="libgazebo_ros_link_attacher.so"/>
    <plugin name="follow_object" filename="libfollow_object.so"/>
    <plugin name="move_to_coord" filename="libmove_to_coord.so"/>
    <plugin name="goal_out_detection" filename="libgoal_out_detection.so"/>
    <plugin name="game_plugin" filename="libgame_plugin.so"/>
    <plugin name="find_distance" filename="libfind_distance.so"/>
    <!-- A ground plane -->
    <include>
      <uri>model://ground_plane</uri>
    </include>
    <!-- A global light source-->
    <include>
      <uri>model://sun</uri>
    </include>
    <!-- RoboCup SPL 2014 field -->
    <include>
      <uri>model://robocup_ssl_field</uri>
    </include>
    <!-- RoboCup SPL 2014 left goal -->
    <include>
      <pose>-4.5 0 0 0 0 3.1416</pose>
      <uri>model://robocup_ssl_goal</uri>
      <name>robocup_ssl_left_goal</name>
    </include>
    <!-- RoboCup SPL 2014 right goal -->
    <include>
      <pose>4.5 0 0 0 0 0</pose>
      <uri>model://robocup_ssl_goal</uri>
      <name>robocup_ssl_right_goal</name>
    </include>
    <!-- RoboCup SPL ball -->
    <include>
      <pose>0 0 0.0315 0 0 0</pose>
      <uri>model://robocup_ssl_ball</uri>
      <name>ball</name>
      <plugin name="ball_controller" filename="libball_controller.so"/>
    </include>
    <!-- A_robot2 -->
    <include>
      <pose>-2 0 0 0 0 3.1416</pose>
      <uri>model://robocup_ssl_robot</uri>
      <!-- <uri>model://turtlebot</uri> -->
      <name>A_robot1</name>
      <plugin name="behavior_tree" filename="libbehavior_tree.so"/>
    </include>
    <!-- A_robot2 -->
    <include>
      <pose>-1.5 1 0 0 0 3.1416</pose>
      <uri>model://robocup_ssl_robot</uri>
      <!-- <uri>model://turtlebot</uri> -->
      <name>A_robot2</name>
      <plugin name="behavior_tree" filename="libbehavior_tree.so"/>
    </include>
    <!-- A_robot3 -->
    <include>
      <pose>-1.5 -1 0 0 0 3.1416</pose>
      <uri>model://robocup_ssl_robot</uri>
      <!-- <uri>model://turtlebot</uri> -->
      <name>A_robot3</name>
      <plugin name="behavior_tree" filename="libbehavior_tree.so"/>
    </include>
    <!-- B_robot2 -->
    <include>
      <pose>-0.5 0 0 0 0 0</pose>
      <uri>model://robocup_ssl_robot</uri>
      <!-- <uri>model://turtlebot</uri> -->
      <name>B_robot1</name>
      <plugin name="behavior_tree" filename="libbehavior_tree.so"/>
    </include>
    <!-- B_robot2 -->
    <include>
      <pose>1.5 1 0 0 0 0</pose>
      <uri>model://robocup_ssl_robot</uri>
      <!-- <uri>model://turtlebot</uri> -->
      <name>B_robot2</name>
      <plugin name="behavior_tree" filename="libbehavior_tree.so"/>
    </include>
    <!-- B_robot3 -->
    <include>
      <pose>1.5 -1 0 0 0 0</pose>
      <uri>model://robocup_ssl_robot</uri>
      <!-- <uri>model://turtlebot</uri> -->
      <name>B_robot3</name>
      <plugin name="behavior_tree" filename="libbehavior_tree.so"/>
    </include>
  </world>
</sdf>